direct/nn/vsharp/vsharp_engine.py

```diff
@@ -18,6 +18,134 @@
 from direct.utils import detach_dict, dict_to_device
 
 
+class VSharpNet3DEngine(MRIModelEngine):
+    """VSharpNet Engine."""
+
+    def __init__(
+        self,
+        cfg: BaseConfig,
+        model: nn.Module,
+        device: str,
+        forward_operator: Optional[callable] = None,
+        backward_operator: Optional[callable] = None,
+        mixed_precision: bool = False,
+        **models: nn.Module,
+    ):
+        """Inits :class:`VSharpNetEngine`.
+
+        Parameters
+        ----------
+        cfg: BaseConfig
+            Configuration file.
+        model: nn.Module
+            Model.
+        device: str
+            Device. Can be "cuda:{idx}" or "cpu".
+        forward_operator: callable, optional
+            The forward operator. Default: None.
+        backward_operator: callable, optional
+            The backward operator. Default: None.
+        mixed_precision: bool
+            Use mixed precision. Default: False.
+        **models: nn.Module
+            Additional models.
+        """
+        super().__init__(
+            cfg,
+            model,
+            device,
+            forward_operator=forward_operator,
+            backward_operator=backward_operator,
+            mixed_precision=mixed_precision,
+            **models,
+        )
+
+        self._spatial_dims = (3, 4)
+
```

Comment:
If have the feeling this is done slightly differently elsewhere, no?

Reply:
No for engines this is how it's defined.